fullstory-segment-middleware/src/test/java/com/fullstorydev/fullstory_segment_middleware/FSPropertiesUtils.java

```diff
@@ -0,0 +1,388 @@
+package com.fullstorydev.fullstory_segment_middleware;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FSPropertiesUtils {
+    static Map<String, Object> ECommerceEventsProductsSearched() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("query_str","blue roses");
+        return map;
+    }
+
+    static Map<String, Object> ECommerceEventsProductListViewed() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("list_id_str","hot_deals_1");
+        map.put("category_str","Deals");
+
+        map.putAll(getGameProducts());
+
+        return map;
+    }
+
+    static Map<String, Object> ECommerceEventsProductListFiltered() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("list_id_str","todays_deals_may_11_2019");
+        map.put("filters.type_strs",new ArrayList<>(Arrays.asList("department","price")));
```

Comment:
It looks like a couple of things are happening here:
The ecommerce spec structure is being "flattened" - for example:
```
filters: { type: 'whatever' }
```

is being converted into a Key Value Pair using and object path string as the key: 

```
("filters.type_strs", "whatever")
```

and

the type suffix is being added to the property: `type --> type_str`

Is there a way to test these two things in isolation?

Reply:
Yup, I had the utilities classes follow exactly the segment spec.
Right now we don't have unit test for a lot of the methods yet, when we add to the test following methods:
addSimpleObjectToMap and pluralizeAllArrayKeysInMap we should basically have separated testing for above scenarios